python/TestHarness/schedulers/Scheduler.py

```diff
@@ -0,0 +1,491 @@
+                tester_dag.add_node(tester.getTestName())
+
+            # Loop through prereqs and add leaves/nodes where necessary
+            for prereq in tester.getPrereqs():
+
+                # Discover unknown prereqs
+                if prereq not in [x.getTestName() for x in testers]:
+                    tester.setStatus('unknown dependency', tester.bucket_fail)
+                    skipped_or_failed_tests.add(tester_data)
+                    continue
+
+                if not tester_dag.node_exists(prereq):
+                    # Create a node for this tester
+                    tester_dag.add_node(prereq)
+
+                # Once you add an edge to a node, the DAG arranges the nodes accordingly
+                try:
+                    tester_dag.add_edge(prereq, tester.getTestName())
+
+                # Handle Cyclic errors
+                except dag.DAGValidationError:
+                    tester.setStatus('Cyclic or Invalid Dependency Detected!', tester.bucket_fail)
+                    skipped_or_failed_tests.add(tester_data)
+
+                    # Pass so we allow the status runner to print the results serially
+                    pass
+
+            # Discover skipped tests
+            if not tester.getRunnable(self.options):
+                skipped_or_failed_tests.add(tester_data)
+
+        # Discover skipped tests and remove downstream tests that require this skipped test. Any
+        # discovered downstream nodes will have 'skipped dependency' set as their status.
+        for skipped_tester_data in skipped_or_failed_tests:
+            tmp_tester = skipped_tester_data.getTester()
+
+            # Delete downstream nodes
+            self.deleteDownstreamTests(skipped_tester_data)
+
+            # Delete this node
+            tester_dag.delete_node_if_exists(tmp_tester.getTestName())
+
+            # Inform the user this node was skipped by adding it to the status queue
+            self.assignStatus(tmp_tester.getTestName())
+
+        # Is there anything to do? All tests were skipped?
+        if tester_dag.size() == 0:
+            return
+
+        # Get top level nodes we can launch now
+        ind_jobs = tester_dag.ind_nodes()
+
+        # Set the status on all the nodes (jobs) this method is about to launch as QUEUED before we actually
+        # launch them so as to prevent a very fast finishing job from asking for the next available job(s) this
+        # method is about to launch (in other words; the status runner launches jobs that are set as INITIALIZED
+        # and not already QUEUED while this method makes no comparison and 'just goes')
+        for job in ind_jobs:
+            tester = self.scheduled_tester_data[job].getTester()
+            tester.setStatus('QUEUED', tester.bucket_pending)
+
+        # With the job properly status'd above... launch them all asynchronously
+        for job in ind_jobs:
+            self.assignRunner(job)
```

Comment:
I think your assignRunner() routine should take a list of items (something iterable). Inside that method you'll perform the loop. Still don't like the methodName "assignRunner" the reason is that you really aren't assigning a runner. You don't really have the authority to do that, that's technically the ThreadPool's job. Instead you are queuing jobs to run. 

How about you make a routine that takes both the "list" and the "ThreadPool" object or a name and you call the method `queueJobs()`